### PR TITLE
chore: add FeatureManagement documentation

### LINKED ISSUE #188: Add Microsoft FeatureManagement Documentation
There's a [Mircosoft FeatureManagement provider](https://github.com/open-feature/dotnet-sdk-contrib/tree/main/src/OpenFeature.Contrib.Providers.FeatureManagement) but it doesn't have documentation. Without docs, it's difficult for new users to learn how the provider works. A readme should be created under `/src/OpenFeature.Contrib.Providers.FeatureManagement` that explains how the add the dependency to the project, how to use it in openfeature, and any limitations or gotchas that a user may encounter. Once the docs have been updated, this provider can be added to the [OpenFeature ecosystem](https://openfeature.dev/ecosystem) page.

### PR DESCRIPTION
**Pull request by ericpattison**
## This PR
Adding in a README.md for the FeatureManagement Provider

- Initial Documentation showing usage of the FeatureManagement Provider

### Related Issues

Fixes #188 

### Follow-up Tasks
FeatureManagement support is still in Preview. When a full release is integrated, the README should be updated accordingly.

### CODE REVIEW

File: src/OpenFeature.Contrib.Providers.FeatureManagement/README.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
+# FeatureManagement .NET Provider
+> **Note**  
+This requires a new feature of the FeatureManagement system, Variants. This feature is still in preview and has not been fully released.
+
+The FeatureManagement Provider allows you to use the FeatureManagement system as an OpenFeature Provider.
+
+# .NET SDK Usage
```
Comment (beeme1mr) [inline src/OpenFeature.Contrib.Providers.FeatureManagement/README.md:7]: Could you please make this an h2 and bump all the titles below by one?

File: src/OpenFeature.Contrib.Providers.FeatureManagement/README.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+## Install dependencies
+
+### .NET Cli
+
+```shell
+dotnet add package OpenFeature.Contrib.Provider.FeatureManagement --version 0.0.1-preview
```
Comment (beeme1mr) [inline src/OpenFeature.Contrib.Providers.FeatureManagement/README.md:14]: Could you update the Release Please configuration to bump the versions? To do that, you'll need to add the README to the [Release Please config](https://github.com/open-feature/dotnet-sdk-contrib/blob/main/release-please-config.json#L62) and update the README with the [appropriate comments](https://github.com/googleapis/release-please/blob/main/docs/customizing.md#updating-arbitrary-files).

File: src/OpenFeature.Contrib.Providers.FeatureManagement/README.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
+# FeatureManagement .NET Provider
+> **Note**  
+This requires a new feature of the FeatureManagement system, Variants. This feature is still in preview and has not been fully released.
```
Comment (beeme1mr) [inline src/OpenFeature.Contrib.Providers.FeatureManagement/README.md:3]:
```suggestion
> ![NOTE]  
> This requires a new feature of the FeatureManagement system, Variants. This feature is still in preview and has not been fully released.
```

https://github.com/orgs/community/discussions/16925

**Review by toddbaert — COMMENTED** [review]
I approve in theory but agree with @beeme1mr 's comments.

Thanks @ericpattison

File: src/OpenFeature.Contrib.Providers.FeatureManagement/README.md (hunk 2 above)
Comment (ericpattison) [reply src/OpenFeature.Contrib.Providers.FeatureManagement/README.md:15]: Added to release please.

File: src/OpenFeature.Contrib.Providers.FeatureManagement/README.md (hunk 1 above)
Comment (ericpattison) [reply src/OpenFeature.Contrib.Providers.FeatureManagement/README.md]: Adjusted the sections.

**Review by beeme1mr — APPROVED** [review]

**Pull request merged**